File: ocaml/triangle/triangle.ml
let is_triangle a b c = 
  max a (max b c) <  a + b + c - (max a (max b c))

let is_equilateral a b c = 
  (a = b && b = c) && is_triangle a b c

let is_isosceles a b c = 
  (a = b || b = c || c = a) && is_triangle a b c

let is_scalene a b c =  
  not (is_isosceles a b c) && is_triangle a b c

File: julia/triangle/triangle.jl
is_triangle(sides) = length(sides) == 3 && minimum(sides) > 0 && sort(sides) |> a -> a[3] <= a[1] + a[2]
unique_values(xs, comp, x) = comp(xs |> Set |> length, x)

is_equilateral(sides) = is_triangle(sides) && unique_values(sides, (==), 1)
is_isosceles(sides)   = is_triangle(sides) && unique_values(sides, (<=), 2)
is_scalene(sides)     = is_triangle(sides) && unique_values(sides, (==), 3)

